code/mtg.cs

```diff
@@ -237,6 +241,48 @@ public List<int> Roots(int scale = 0)
 
         #endregion
 
+        #region Complex
+
+        /// <summary>
+        /// Gets the complex value for a key.
+        /// </summary>
+        /// <param name="vertexId"> Vertex identifier.</param>
+        /// <returns> Returns the vertex's complex. If it doesn't have a complex, it returns null. </returns>
+        int? GetComplex(int vertexId)
+        {
+            if (complex.ContainsKey(vertexId))
+                return complex[vertexId];
+            else
+                return null;
+        }
+
+        /// <summary>
+        /// Returns the complex of a vertex.
+        /// </summary>
+        /// <param name="vertexId"> Vertex identifier. </param>
+        /// <returns> The complex of the vertex or null. </returns>
+        public int? Complex(int vertexId)
+        {
```

Comment:
@pradal : In case the vertex doesn't have a complex, should it return null or a default value ?